### DISCUSSION TITLE
failover filer server

### DISCUSSION THREAD

**Discussion by aoberest** [seaweedfs/seaweedfs#1859] [Q&A] [answered]
Hello. I am testing  failover filer server.
weed version
version 30GB 2.26 71f0c19 linux amd64

I am running cluster seaweedfs with 3 masters, 2 filers, 6 volumes. 
server1 - master,volume, filer
server2 - master,volume, filer
server3 - master,volume
server4-sever6 - only volumes
Cluster worked good. I loaded data into two s3 buckets: bucket1 and bucket2.
It showed all this buckets in filer1 and filer2.
The filers use local leveldb3 stores.
Both of filers have active replication (filer1 <-> filer2 )

Example start 'filer2' 
```
/usr/bin/weed filer -s3 -s3.config=/var/lib/seaweedfs/filer/config.json -defaultStoreDir=/var/lib/seaweedfs/filer -ip=server2 -port=8888 -dataCenter=stage  -peers=server1:8888,server2:8888  -master=server1:9333,server2:9333,server3:9333
```

Then I deleted all data from server1 (master1, volume1, filer1 metadata) and started all services again.

On the second filer, I could still see all the buckets, but on the first filer, there were no buckets.  
I waited for over an hour, but nothing changed. 
I didn't find any errors in the logs. 
Then created bucket1 in filer1 and uploaded one file and saw it in filer1 and filer2.

Should the filer have been replicated automatically after crashes or loss of meta data on one of the filer ?

What are the correct steps I should take to recover the filer1 metadata ? 
Can they be replicated from the working filer2 ?

**Comment by chrislusf** [comment, marked as answer]
This needs a step to bootstrap from one of the other peers. Will add this feature later.

For now, with the latest code, you can run `weed filer.meta.backup` to make a copy of the metadata from the other peer, and start the filer.

**Reply by aoberest** [reply]
Thank you for answering my question.

It would be nice if it was possible to check the consistency of the metadata on the filers and synchronize them. 
Maybe this could be done automatically or by running the command on a schedule.

**Reply by chrislusf** [reply]
> This needs a step to bootstrap from one of the other peers. Will add this feature later.

added https://github.com/chrislusf/seaweedfs/issues/1861

**Reply by aoberest** [reply]
```
#  weed filer.meta.backup
weed: unknown subcommand "filer.meta.backup"
Run 'weed help' for usage.
```
```
# weed version
version 30GB 2.29 f29ae0d linux amd64
```

Please tell me how to make a copy of the metadata ?

**Reply by chrislusf** [reply]
use the dev version https://github.com/chrislusf/seaweedfs/releases/tag/dev

**Comment by ghost** [comment]
Not sure if I should start a new discussion for this - let me know if that's preferred. 

This is similar to the ideas in #1531, #2070 and others.
- If I deploy seaweedfs with multiple masters (on different hosts), multiple volume services (on different hosts).. what about my filer? 
- If I have just 1 filer, it's my SPOF. If I have several filers.. how do I keep them synchronized? (seems to be a lot of active issues around this)

What's the best practice right now for having HA filers?

For some use cases (like mine) the goal is to have no SPOF. Not just to survive random failures, but also to enable rolling maintenance work on the hosts. The wiki [mentions google's single master system](https://github.com/chrislusf/seaweedfs/wiki/Failover-Master-Server) but I'm not sure if it deals with this..

**Reply by chrislusf** [reply]
fix the issues?

The safer setup can be using one shared filer store, and multiple stateless filer instances.

**Reply by aoberest** [reply]
> Not sure if I should start a new discussion for this - let me know if that's preferred.
> 
> This is similar to the ideas in #1531, #2070 and others.
> 
> * If I deploy seaweedfs with multiple masters (on different hosts), multiple volume services (on different hosts).. what about my filer?
> * If I have just 1 filer, it's my SPOF. If I have several filers.. how do I keep them synchronized? (seems to be a lot of active issues around this)
> 
> What's the best practice right now for having HA filers?
> 
> For some use cases (like mine) the goal is to have no SPOF. Not just to survive random failures, but also to enable rolling maintenance work on the hosts. The wiki [mentions google's single master system](https://github.com/chrislusf/seaweedfs/wiki/Failover-Master-Server) but I'm not sure if it deals with this..

Hello.

I use 3 filers and one shared Postgresql metadata store.
I made a dns record of seaweed-cluster.main.corp and added the IP addresses of all the "filers". 
https://en.wikipedia.org/wiki/Round-robin_DNS
